### PR TITLE
Add Cell class

### PR DESCRIPTION
**Pull request by ajtran303**
Add Cell class

### CODE REVIEW

File: lib/cell.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      "H"
+    elsif !empty? && reveal == true
+      "S"
+    elsif empty? && fired_upon? || reveal == true
+      "M"
+    elsif !fired_upon?
+      "."
+    end
+  end
+
```
Comment (ajtran303) [inline lib/cell.rb:41]: This can be refactored later.

Comment (GDemps) [reply lib/cell.rb:41]: Good idea.

File: lib/cell.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    @ship == nil
+  end
+
+  def place_ship(ship_obj)
+    @ship = ship_obj
+  end
+
+  def fire_upon
+    @is_fired_upon = true
+    if !empty?
```
Comment (GDemps) [inline lib/cell.rb:21]: Good use of the bang to call the inverse!

**Review by GDemps — APPROVED** [review]
Good job with the Cell class and testing.

**Pull request merged**